vulture_os/toolkit/api_parser/trendmicro_visionone/trendmicro_visionone.py

```diff
@@ -133,57 +181,60 @@ def _format_logs(log):
     def execute(self):
         for kind in ["alerts", "oat", "audit"]:
 
-            try:
-                since = getattr(self.frontend, f"trendmicro_visionone_{kind}_timestamp") or (timezone.now() - timedelta(days=2))
-                to = since + timedelta(minutes=1)
+            since = getattr(self.frontend, f"trendmicro_visionone_{kind}_timestamp") or (
+                        timezone.now() - timedelta(days=2))
+            to = min(timezone.now() - timedelta(minutes=5), since + timedelta(hours=24))
 
-                while to <= timezone.now() - timedelta(minutes=5) and not self.evt_stop.is_set():
-                    logger.info(f"[{__parser__}]:execute: Parser gets {kind} logs from {since} to {to}", extra={'frontend': str(self.frontend)})
-                    start_time = since.strftime("%Y-%m-%dT%H:%M:%SZ")
-                    end_time = to.strftime("%Y-%m-%dT%H:%M:%SZ")
+            while to <= timezone.now() - timedelta(minutes=5) and not self.evt_stop.is_set():
```

Comment:
I see a world where the collector is struggling to get to that point, and the 2 other types of logs are never handled (or with great delay), could you try reworking this `for..while` structure that existed to avoid that problem?